From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Tue, 2 Feb 2021 17:29:29 +0100
Subject: gdm: Ensure we cancel all the previously running services on auth
 retry

When retrying the authentication we should make sure that all the
previously initiated services are stopped in order to begin a new
authentication session with all the configured services.

Unfortunately at the current state we only dispose the currently used
user verifier, but we don't make it to stop all the relative gdm workers
and then they'll stay around potentially blocking any further usage of
them (as it happens with the fingerprint one, that has unique access to
the device).

So, cancel the currently running authentication before starting a new
one if we're explicitly retrying.

Origin: https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/1622
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+source/gnome-shell/+bug/1915066
---
 js/gdm/util.js | 1 +
 1 file changed, 1 insertion(+)

diff --git a/js/gdm/util.js b/js/gdm/util.js
index 17377d7..29921fc 100644
--- a/js/gdm/util.js
+++ b/js/gdm/util.js
@@ -548,6 +548,7 @@ var ShellUserVerifier = class {
     }
 
     _retry() {
+        this.cancel();
         this.begin(this._userName, new Batch.Hold());
     }
 
